modules/core/src/main/java/org/apache/ignite/internal/util/GridUnsafe.java

```diff
@@ -1587,17 +1595,31 @@ private static Object javaNioAccessObject() {
      * @return Reference to {@code JavaNioAccess.newDirectByteBuffer} method
      * @throws RuntimeException If getting access to the private API is failed.
      */
-    private static Method newDirectBufferMethod(Object nioAccessObj) {
+    private static MethodHandle newDirectBufferMethodHandle(Object nioAccessObj) {
         try {
             Class<?> cls = nioAccessObj.getClass();
 
             Method mtd = cls.getMethod("newDirectByteBuffer", long.class, int.class, Object.class);
 
-            mtd.setAccessible(true);
+            AccessController.doPrivileged((PrivilegedExceptionAction<?>)() -> {
+                mtd.setAccessible(true);
+
+                return null;
+            });
 
-            return mtd;
+            MethodType mtdType = MethodType.methodType(
+                ByteBuffer.class,
+                Object.class,
+                long.class,
+                int.class,
+                Object.class
+            );
+
+            return MethodHandles.lookup()
+                .unreflect(mtd)
+                .asType(mtdType);
```

Comment:
why is this line needed?

Reply:
You mean "asType"? For "invokeExact" to work properly

Reply:
Rechecked, code won't work without explicit type:
java.lang.invoke.WrongMethodTypeException: expected (,long,int,Object)ByteBuffer but found (Object,long,int,Object)ByteBuffer